File: spring-practice/src/main/java/com/example/springpractice/order/dto/OrderResponse.java
package com.example.springpractice.order.dto;

import com.example.springpractice.order.Order;
import com.example.springpractice.order.OrderItem;
import java.time.LocalDateTime;
import java.util.List;

public record OrderResponse(
        Long orderId,
        String ordererName,
        LocalDateTime orderedAt,
        List<OrderItemResponse> items,
        int totalAmount
) {
    public static OrderResponse from(Order order) {
        List<OrderItemResponse> items = order.getOrderItems().stream()
                .map(OrderItemResponse::from)
                .toList();
        return new OrderResponse(
                order.getId(),
                order.getOrdererName(),
                order.getOrderedAt(),
                items,
                order.getTotalAmount()
        );
    }

    public record OrderItemResponse(
            Long productId,
            String productName,
            int quantity,
            int orderPrice,
            int subtotal
    ) {
        public static OrderItemResponse from(OrderItem orderItem) {
            return new OrderItemResponse(
                    orderItem.getProduct().getId(),
                    orderItem.getProduct().getName(),
                    orderItem.getQuantity(),
                    orderItem.getOrderPrice(),
                    orderItem.getSubtotal()
            );
        }
    }
}

File: kotlin-practice/src/main/kotlin/com/example/kotlinpractice/order/dto/OrderResponse.kt
package com.example.kotlinpractice.order.dto

import com.example.kotlinpractice.order.Order
import com.example.kotlinpractice.order.OrderItem
import java.time.LocalDateTime

data class OrderResponse(
    val orderId: Long?,
    val ordererName: String,
    val orderedAt: LocalDateTime,
    val items: List<OrderItemResponse>,
    val totalAmount: Int
) {
    data class OrderItemResponse(
        val productId: Long?,
        val productName: String,
        val quantity: Int,
        val orderPrice: Int,
        val subtotal: Int
    ) {
        companion object {
            fun from(orderItem: OrderItem): OrderItemResponse =
                OrderItemResponse(
                    orderItem.product.id,
                    orderItem.product.name,
                    orderItem.quantity,
                    orderItem.orderPrice,
                    orderItem.subtotal
                )
        }
    }

    companion object {
        fun from(order: Order): OrderResponse {
            val items = order.orderItems.map { OrderItemResponse.from(it) }
            return OrderResponse(
                order.id,
                order.ordererName,
                order.orderedAt,
                items,
                order.totalAmount
            )
        }
    }
}
